homeassistant/config_entries.py

```diff
@@ -1884,8 +1890,16 @@ async def async_forward_entry_setup(
             if not result:
                 return False
 
-        integration = await loader.async_get_integration(self.hass, domain)
+        if preload_platform:
+            # If this is a late setup, we need to make sure the platform is loaded
+            # so we do not end up waiting for when the EntityComponent calls
+            # async_prepare_setup_platform
+            integration = await loader.async_get_integration(self.hass, entry.domain)
```

Comment:
If some integration forwards the platform setup multiple times the numbers will go negative here, but that seems ok since that would be a bug in the integration

Reply:
This actually can happen in other cases, sometimes there is a unload,reload of a platform because something changes, or a late load of a platform created in a background task. So we need to handle the case where the pause finishes and the running setup that it was paused in was already done.  Sadly that adds more complexity.  I'm going to try to reduce it down a bit

Reply:
It turned out be a simple check of if the group is still in `_setup_started`. Thankfully I was able to remove all the additional complexity I added when I was testing out solutions here